{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import nltk\n",
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "from tqdm import tqdm\n",
    "import glob\n",
    "import nltk\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.tokenize import word_tokenize\n",
    "import os\n",
    "import json\n",
    "import warnings\n",
    "from collections import Counter\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from nltk.corpus import reuters\n",
    "from nltk.stem.snowball import SnowballStemmer\n",
    "from nltk.corpus import stopwords\n",
    "import nltk.data\n",
    "import math\n",
    "import re\n",
    "from nltk import word_tokenize, pos_tag\n",
    "from nltk.corpus import wordnet as wn\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scipy as sc\n",
    "\n",
    "import os\n",
    "import json\n",
    "import warnings\n",
    "\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "DEBUG = False\n",
    "SUMMARY_LENGTH = 12  # number of sentences in final summary\n",
    "stop_words = stopwords.words('english')\n",
    "ideal_sent_length = 20.0\n",
    "stemmer = SnowballStemmer(\"english\")\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "import re\n",
    "from urllib.request import urlopen\n",
    "#import gensim \n",
    "import numpy as np \n",
    "import pandas as pd\n",
    "\n",
    "from sklearn.feature_extraction import text\n",
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer\n",
    "from sklearn.decomposition import LatentDirichletAllocation\n",
    "\n",
    "import pandas as pd  # For data handling\n",
    "from time import time  # To time our operations\n",
    "from collections import defaultdict  # For word frequency\n",
    "import re\n",
    "import spacy  # For preprocessing\n",
    "\n",
    "import logging  # Setting up the loggings to monitor gensim\n",
    "\n",
    "import spacy\n",
    "import joblib\n",
    "\n",
    "\n",
    "from IPython.display import HTML, display\n",
    "from ipywidgets import interact, Layout, HBox, VBox, Box\n",
    "import ipywidgets as widgets\n",
    "from IPython.display import clear_output\n",
    "\n",
    "from tqdm import tqdm\n",
    "\n",
    "import seaborn as sb\n",
    "import matplotlib.pyplot as plt\n",
    "plt.style.use(\"dark_background\")\n",
    "import nltk\n",
    "from nltk.corpus import wordnet as wn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15758"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = r\"C:\\Users\\ThinkPad\\OneDrive\\Bureau\\CORD-19-research-challenge\"\n",
    "all_json = glob.glob(f'{path}/**/*.json', recursive=True)\n",
    "len(all_json)\n",
    "metadata_path = f'{path}/metadata.csv'\n",
    "# load metadata\n",
    "metadata = pd.read_csv(metadata_path)\n",
    "metadata.head()\n",
    "metadata.sha.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3...\n"
     ]
    }
   ],
   "source": [
    "class FileReader:\n",
    "    def __init__(self, file_path):\n",
    "        with open(file_path) as file:\n",
    "            dic={'title':[]} \n",
    "            content = json.load(file)\n",
    "            self.paper_id = content['paper_id']\n",
    "            self.metadataa = []\n",
    "            self.metadata = content['metadata']\n",
    "            meta=self.metadata\n",
    "            self.abstract = []\n",
    "            self.body_text = []\n",
    "            \n",
    "            \n",
    "           \n",
    "            # Abstract\n",
    "            for entry in content['abstract']:\n",
    "                self.abstract.append(entry['text'])\n",
    "            # Body text\n",
    "            for entry in content['body_text']:\n",
    "                self.body_text.append(entry['text'])\n",
    "                \n",
    "           \n",
    "                tit=meta['title']\n",
    "                dic['title'].append(tit)\n",
    "            self.abstract = '\\n'.join(self.abstract)\n",
    "            self.body_text = '\\n'.join(self.body_text)\n",
    "            self.metadataa = tit\n",
    "    def __repr__(self):\n",
    "        return f'{self.metadataa}...'\n",
    "first_row = FileReader(all_json[0])\n",
    "print(first_row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing index: 0 of 29315\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>body_text</th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0015023cc06b5362d332b3baf348d11567ca2fbb</td>\n",
       "      <td>word count: 194 22 Text word count: 5168 23 24...</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP...</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>004f0f8bb66cf446678dc13cf2701feec4f36d76</td>\n",
       "      <td></td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China...</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00d16927588fb04d4be0e6b269fc02f0d3c2aa7b</td>\n",
       "      <td>Infectious bronchitis (IB) causes significant ...</td>\n",
       "      <td>Infectious bronchitis (IB), which is caused by...</td>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0139ea4ca580af99b602c6435368e7fdbefacb03</td>\n",
       "      <td>Nipah Virus (NiV) came into limelight recently...</td>\n",
       "      <td>Nipah is an infectious negative-sense single-s...</td>\n",
       "      <td>A Combined Evidence Approach to Prioritize Nip...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>013d9d1cba8a54d5d3718c229b812d7cf91b6c89</td>\n",
       "      <td>Background: A novel coronavirus (2019-nCoV) em...</td>\n",
       "      <td>In December 2019, a cluster of patients with p...</td>\n",
       "      <td>Assessing spread risk of Wuhan novel coronavir...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   paper_id  \\\n",
       "0  0015023cc06b5362d332b3baf348d11567ca2fbb   \n",
       "1  004f0f8bb66cf446678dc13cf2701feec4f36d76   \n",
       "2  00d16927588fb04d4be0e6b269fc02f0d3c2aa7b   \n",
       "3  0139ea4ca580af99b602c6435368e7fdbefacb03   \n",
       "4  013d9d1cba8a54d5d3718c229b812d7cf91b6c89   \n",
       "\n",
       "                                            abstract  \\\n",
       "0  word count: 194 22 Text word count: 5168 23 24...   \n",
       "1                                                      \n",
       "2  Infectious bronchitis (IB) causes significant ...   \n",
       "3  Nipah Virus (NiV) came into limelight recently...   \n",
       "4  Background: A novel coronavirus (2019-nCoV) em...   \n",
       "\n",
       "                                           body_text  \\\n",
       "0  VP3, and VP0 (which is further processed to VP...   \n",
       "1  The 2019-nCoV epidemic has spread across China...   \n",
       "2  Infectious bronchitis (IB), which is caused by...   \n",
       "3  Nipah is an infectious negative-sense single-s...   \n",
       "4  In December 2019, a cluster of patients with p...   \n",
       "\n",
       "                                               title  \n",
       "0  The RNA pseudoknots in foot-and-mouth disease ...  \n",
       "1  Healthcare-resource-adjusted vulnerabilities t...  \n",
       "2  Real-time, MinION-based, amplicon sequencing f...  \n",
       "3  A Combined Evidence Approach to Prioritize Nip...  \n",
       "4  Assessing spread risk of Wuhan novel coronavir...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_ = {'paper_id': [], 'abstract': [], 'body_text': [],'title':[]}\n",
    "for idx, entry in enumerate(all_json[0:1000]):\n",
    "    if idx % (len(all_json) // 10) == 0:\n",
    "        print(f'Processing index: {idx} of {len(all_json)}')\n",
    "    content = FileReader(entry)\n",
    "    dict_['paper_id'].append(content.paper_id)\n",
    "    dict_['abstract'].append(content.abstract)\n",
    "    dict_['body_text'].append(content.body_text)\n",
    "    dict_['title'].append(content.metadataa)\n",
    "    \n",
    "    \n",
    "papers = pd.DataFrame(dict_, columns=['paper_id', 'abstract', 'body_text','title'])\n",
    "papers.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1000, 4)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# any true nulls in the data?\n",
    "papers.isnull().sum(axis=0)\n",
    "# a few dupes - get rid of those\n",
    "papers.drop_duplicates('body_text', inplace=True)\n",
    "papers.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>body_text</th>\n",
       "      <th>title</th>\n",
       "      <th>source_x</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0015023cc06b5362d332b3baf348d11567ca2fbb</td>\n",
       "      <td>word count: 194 22 Text word count: 5168 23 24...</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP...</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>10.1101/2020.01.10.901801</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>2020-01-11</td>\n",
       "      <td>Ward, J. C. J.; Lasecka-Dykes, L.; Neil, C.; A...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>004f0f8bb66cf446678dc13cf2701feec4f36d76</td>\n",
       "      <td></td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China...</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>10.1101/2020.02.11.20022111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>medrvix</td>\n",
       "      <td>2020-02-12</td>\n",
       "      <td>Hanchu Zhou; Jianan Yang; Kaichen Tang; Qingpe...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00d16927588fb04d4be0e6b269fc02f0d3c2aa7b</td>\n",
       "      <td>Infectious bronchitis (IB) causes significant ...</td>\n",
       "      <td>Infectious bronchitis (IB), which is caused by...</td>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing f...</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>10.1101/634600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>2019-05-10</td>\n",
       "      <td>Butt, S. L.; Erwood, E. C.; Zhang, J.; Sellers...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0139ea4ca580af99b602c6435368e7fdbefacb03</td>\n",
       "      <td>Nipah Virus (NiV) came into limelight recently...</td>\n",
       "      <td>Nipah is an infectious negative-sense single-s...</td>\n",
       "      <td>A Combined Evidence Approach to Prioritize Nip...</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>10.1101/2020.03.12.977918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>2020-03-12</td>\n",
       "      <td>Nishi Kumari; Ayush Upadhyay; Kishan Kalia; Ra...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>013d9d1cba8a54d5d3718c229b812d7cf91b6c89</td>\n",
       "      <td>Background: A novel coronavirus (2019-nCoV) em...</td>\n",
       "      <td>In December 2019, a cluster of patients with p...</td>\n",
       "      <td>Assessing spread risk of Wuhan novel coronavir...</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>10.1101/2020.02.04.20020479</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>medrvix</td>\n",
       "      <td>2020-02-05</td>\n",
       "      <td>Shengjie Lai; Isaac Bogoch; Nick Ruktanonchai;...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   paper_id  \\\n",
       "0  0015023cc06b5362d332b3baf348d11567ca2fbb   \n",
       "1  004f0f8bb66cf446678dc13cf2701feec4f36d76   \n",
       "2  00d16927588fb04d4be0e6b269fc02f0d3c2aa7b   \n",
       "3  0139ea4ca580af99b602c6435368e7fdbefacb03   \n",
       "4  013d9d1cba8a54d5d3718c229b812d7cf91b6c89   \n",
       "\n",
       "                                            abstract  \\\n",
       "0  word count: 194 22 Text word count: 5168 23 24...   \n",
       "1                                                      \n",
       "2  Infectious bronchitis (IB) causes significant ...   \n",
       "3  Nipah Virus (NiV) came into limelight recently...   \n",
       "4  Background: A novel coronavirus (2019-nCoV) em...   \n",
       "\n",
       "                                           body_text  \\\n",
       "0  VP3, and VP0 (which is further processed to VP...   \n",
       "1  The 2019-nCoV epidemic has spread across China...   \n",
       "2  Infectious bronchitis (IB), which is caused by...   \n",
       "3  Nipah is an infectious negative-sense single-s...   \n",
       "4  In December 2019, a cluster of patients with p...   \n",
       "\n",
       "                                               title source_x  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease ...  biorxiv   \n",
       "1  Healthcare-resource-adjusted vulnerabilities t...  medrxiv   \n",
       "2  Real-time, MinION-based, amplicon sequencing f...  biorxiv   \n",
       "3  A Combined Evidence Approach to Prioritize Nip...  biorxiv   \n",
       "4  Assessing spread risk of Wuhan novel coronavir...  medrxiv   \n",
       "\n",
       "                           doi pmcid  pubmed_id  license publish_time  \\\n",
       "0    10.1101/2020.01.10.901801   NaN        NaN  biorxiv   2020-01-11   \n",
       "1  10.1101/2020.02.11.20022111   NaN        NaN  medrvix   2020-02-12   \n",
       "2               10.1101/634600   NaN        NaN  biorxiv   2019-05-10   \n",
       "3    10.1101/2020.03.12.977918   NaN        NaN  biorxiv   2020-03-12   \n",
       "4  10.1101/2020.02.04.20020479   NaN        NaN  medrvix   2020-02-05   \n",
       "\n",
       "                                             authors journal  \\\n",
       "0  Ward, J. C. J.; Lasecka-Dykes, L.; Neil, C.; A...     NaN   \n",
       "1  Hanchu Zhou; Jianan Yang; Kaichen Tang; Qingpe...     NaN   \n",
       "2  Butt, S. L.; Erwood, E. C.; Zhang, J.; Sellers...     NaN   \n",
       "3  Nishi Kumari; Ayush Upadhyay; Kishan Kalia; Ra...     NaN   \n",
       "4  Shengjie Lai; Isaac Bogoch; Nick Ruktanonchai;...     NaN   \n",
       "\n",
       "   Microsoft Academic Paper ID WHO #Covidence has_full_text   full_text_file  \n",
       "0                          NaN            NaN          True  biorxiv_medrxiv  \n",
       "1                          NaN            NaN          True  biorxiv_medrxiv  \n",
       "2                          NaN            NaN          True  biorxiv_medrxiv  \n",
       "3                          NaN            NaN          True  biorxiv_medrxiv  \n",
       "4                          NaN            NaN          True  biorxiv_medrxiv  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_df = papers\\\n",
    "    .merge(metadata.rename(columns={'sha':'paper_id'}).drop(['abstract','title'], axis=1), \n",
    "           on='paper_id', how='left')\n",
    "\n",
    "full_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Intrauterine virus infections and congenital h...</td>\n",
       "      <td>10.1016/0002-8703(72)90077-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4361535.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract The etiologic basis for the vast majo...</td>\n",
       "      <td>1972-12-31</td>\n",
       "      <td>Overall, James C.</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Coronaviruses in Balkan nephritis</td>\n",
       "      <td>10.1016/0002-8703(80)90355-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6243850.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Cigarette smoking and coronary heart disease: ...</td>\n",
       "      <td>10.1016/0002-8703(80)90356-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7355701.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Friedman, Gary D</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aecbc613ebdab36753235197ffb4f35734b5ca63</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Clinical and immunologic studies in identical ...</td>\n",
       "      <td>10.1016/0002-9343(73)90176-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4579077.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Middle-aged female identical twins, o...</td>\n",
       "      <td>1973-08-31</td>\n",
       "      <td>Brunner, Carolyn M.; Horwitz, David A.; Shann,...</td>\n",
       "      <td>The American Journal of Medicine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Epidemiology of community-acquired respiratory...</td>\n",
       "      <td>10.1016/0002-9343(85)90361-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4014285.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Upper respiratory tract infections ar...</td>\n",
       "      <td>1985-06-28</td>\n",
       "      <td>Garibaldi, Richard A.</td>\n",
       "      <td>The American Journal of Medicine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        sha  source_x  \\\n",
       "0                                       NaN  Elsevier   \n",
       "1                                       NaN  Elsevier   \n",
       "2                                       NaN  Elsevier   \n",
       "3  aecbc613ebdab36753235197ffb4f35734b5ca63  Elsevier   \n",
       "4                                       NaN  Elsevier   \n",
       "\n",
       "                                               title  \\\n",
       "0  Intrauterine virus infections and congenital h...   \n",
       "1                  Coronaviruses in Balkan nephritis   \n",
       "2  Cigarette smoking and coronary heart disease: ...   \n",
       "3  Clinical and immunologic studies in identical ...   \n",
       "4  Epidemiology of community-acquired respiratory...   \n",
       "\n",
       "                            doi pmcid  pubmed_id    license  \\\n",
       "0  10.1016/0002-8703(72)90077-4   NaN  4361535.0  els-covid   \n",
       "1  10.1016/0002-8703(80)90355-5   NaN  6243850.0  els-covid   \n",
       "2  10.1016/0002-8703(80)90356-7   NaN  7355701.0  els-covid   \n",
       "3  10.1016/0002-9343(73)90176-9   NaN  4579077.0  els-covid   \n",
       "4  10.1016/0002-9343(85)90361-4   NaN  4014285.0  els-covid   \n",
       "\n",
       "                                            abstract publish_time  \\\n",
       "0  Abstract The etiologic basis for the vast majo...   1972-12-31   \n",
       "1                                                NaN   1980-03-31   \n",
       "2                                                NaN   1980-03-31   \n",
       "3  Abstract Middle-aged female identical twins, o...   1973-08-31   \n",
       "4  Abstract Upper respiratory tract infections ar...   1985-06-28   \n",
       "\n",
       "                                             authors  \\\n",
       "0                                  Overall, James C.   \n",
       "1  Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...   \n",
       "2                                   Friedman, Gary D   \n",
       "3  Brunner, Carolyn M.; Horwitz, David A.; Shann,...   \n",
       "4                              Garibaldi, Richard A.   \n",
       "\n",
       "                            journal  Microsoft Academic Paper ID  \\\n",
       "0            American Heart Journal                          NaN   \n",
       "1            American Heart Journal                          NaN   \n",
       "2            American Heart Journal                          NaN   \n",
       "3  The American Journal of Medicine                          NaN   \n",
       "4  The American Journal of Medicine                          NaN   \n",
       "\n",
       "  WHO #Covidence  has_full_text  full_text_file  \n",
       "0            NaN          False  custom_license  \n",
       "1            NaN          False  custom_license  \n",
       "2            NaN          False  custom_license  \n",
       "3            NaN           True  custom_license  \n",
       "4            NaN          False  custom_license  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covid_csv = pd.read_csv(r'C:\\Users\\ThinkPad\\OneDrive\\Bureau\\CORD-19-research-challenge\\metadata.csv')\n",
    "covid_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000 rows in the output dataframe\n"
     ]
    }
   ],
   "source": [
    "full_df.drop_duplicates('body_text', inplace=True)\n",
    "print(f\"{full_df.shape[0]} rows in the output dataframe\")\n",
    "\n",
    "full_df.to_csv('covid_corpus_33k.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "pre1 = covid_csv.iloc[:,0:15].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "pre = pd.DataFrame()\n",
    "pre['title'] = pd.concat([pre1['title'], papers['title']], ignore_index=True)\n",
    "pre['abstract'] = pd.concat([pre1['abstract'], papers['abstract']], ignore_index=True)\n",
    "pre['body_text'] = pd.concat([pre1['doi'], papers['body_text']], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "summ = pd.DataFrame()\n",
    "summ['title'] =papers['title']\n",
    "summ['body_text'] = papers['body_text']\n",
    "summ.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>body_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing f...</td>\n",
       "      <td>Infectious bronchitis (IB), which is caused by...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A Combined Evidence Approach to Prioritize Nip...</td>\n",
       "      <td>Nipah is an infectious negative-sense single-s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Assessing spread risk of Wuhan novel coronavir...</td>\n",
       "      <td>In December 2019, a cluster of patients with p...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               title  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease ...   \n",
       "1  Healthcare-resource-adjusted vulnerabilities t...   \n",
       "2  Real-time, MinION-based, amplicon sequencing f...   \n",
       "3  A Combined Evidence Approach to Prioritize Nip...   \n",
       "4  Assessing spread risk of Wuhan novel coronavir...   \n",
       "\n",
       "                                           body_text  \n",
       "0  VP3, and VP0 (which is further processed to VP...  \n",
       "1  The 2019-nCoV epidemic has spread across China...  \n",
       "2  Infectious bronchitis (IB), which is caused by...  \n",
       "3  Nipah is an infectious negative-sense single-s...  \n",
       "4  In December 2019, a cluster of patients with p...  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summ.head(5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "summr = pd.read_csv('summr.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\r\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\r\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\r\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\r\\nMutations were then introduced into the replicon plasmid as described above.\\r\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\r\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\r\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\r\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\r\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\r\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\r\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\r\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint '"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summr['body_text'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summr['title'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "SUMMARY_LENGTH = 12  # number of sentences in final summary\n",
    "newcsv=pd.DataFrame()\n",
    "stop_words = stopwords.words('english')\n",
    "ideal_sent_length = 20.0\n",
    "stemmer = SnowballStemmer(\"english\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>body_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Intrauterine virus infections and congenital h...</td>\n",
       "      <td>10.1016/0002-8703(72)90077-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Coronaviruses in Balkan nephritis</td>\n",
       "      <td>10.1016/0002-8703(80)90355-5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cigarette smoking and coronary heart disease: ...</td>\n",
       "      <td>10.1016/0002-8703(80)90356-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Clinical and immunologic studies in identical ...</td>\n",
       "      <td>10.1016/0002-9343(73)90176-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Epidemiology of community-acquired respiratory...</td>\n",
       "      <td>10.1016/0002-9343(85)90361-4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               title  \\\n",
       "0  Intrauterine virus infections and congenital h...   \n",
       "1                  Coronaviruses in Balkan nephritis   \n",
       "2  Cigarette smoking and coronary heart disease: ...   \n",
       "3  Clinical and immunologic studies in identical ...   \n",
       "4  Epidemiology of community-acquired respiratory...   \n",
       "\n",
       "                      body_text  \n",
       "0  10.1016/0002-8703(72)90077-4  \n",
       "1  10.1016/0002-8703(80)90355-5  \n",
       "2  10.1016/0002-8703(80)90356-7  \n",
       "3  10.1016/0002-9343(73)90176-9  \n",
       "4  10.1016/0002-9343(85)90361-4  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new= pd.DataFrame()\n",
    "new['title'] = pre['title']\n",
    "new['body_text'] = pre['body_text']\n",
    "new.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "from nltk.corpus import stopwords\n",
    "stop_words = stopwords.words('english')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Summarizer():\n",
    "    \n",
    "    \n",
    "    def penn_to_wn(self,tag):\n",
    "        \"\"\" Convert between a Penn Treebank tag to a simplified Wordnet tag \"\"\"\n",
    "        if tag.startswith('N'):\n",
    "            return 'n'\n",
    " \n",
    "        if tag.startswith('V'):\n",
    "            return 'v'\n",
    " \n",
    "        if tag.startswith('J'):\n",
    "            return 'a'\n",
    " \n",
    "        if tag.startswith('R'):\n",
    "            return 'r'\n",
    "        return None\n",
    " \n",
    "\n",
    " \n",
    "    def tagged_to_synset(self,word, tag):\n",
    "        wn_tag = self.penn_to_wn(tag)\n",
    "    \n",
    "        if wn_tag is None:\n",
    "            return None\n",
    " \n",
    "        try:\n",
    "        \n",
    "            return wn.synsets(word, wn_tag)[0]\n",
    "        except:\n",
    "            return None\n",
    " \n",
    "    def sentence_similarity(self,sentence1, sentence2):\n",
    "        \"\"\" compute the sentence similarity using Wordnet \"\"\"\n",
    "        # Tokenize and tag\n",
    "        sentence1 = pos_tag(word_tokenize(sentence1))\n",
    "        \n",
    "        sentence2 = pos_tag(word_tokenize(sentence2))\n",
    "      \n",
    " \n",
    "        # Get the synsets for the tagged words\n",
    "        synsets1 = [self.tagged_to_synset(*tagged_word) for tagged_word in sentence1]\n",
    "        \n",
    "        synsets2 = [self.tagged_to_synset(*tagged_word) for tagged_word in sentence2]\n",
    "       \n",
    " \n",
    "    # Filter out the Nones\n",
    "        synsets1 = [ss for ss in synsets1 if ss]\n",
    "      \n",
    "        synsets2 = [ss for ss in synsets2 if ss]\n",
    "        \n",
    " \n",
    "        score, count = 0, 0\n",
    "        for synset in synsets1:\n",
    "            max_sim = 0.0\n",
    "            maxx=0\n",
    "            for ss in synsets2:\n",
    "                \n",
    "                sim=wn.path_similarity(synset, ss)\n",
    "                if sim is not None and sim > max_sim:\n",
    "                \n",
    "                       max_sim = sim\n",
    "                   \n",
    "           \n",
    "        \n",
    "            if max_sim is not None and max_sim!=0:\n",
    "                score += max_sim\n",
    "                count += 1\n",
    " \n",
    "    # Average the values\n",
    "        if count!=0:\n",
    "            score /= count\n",
    "            return score\n",
    "   \n",
    "    def __init__(self, article):\n",
    "            self._articles = []\n",
    "            i=1\n",
    "           \n",
    "            for row in article:\n",
    "                if i<=1000:\n",
    "                    title=row[1]\n",
    "                    #print(title)\n",
    "                    body=row[2].replace('\\n', ' ')\n",
    "                    #print(body)\n",
    "                    \n",
    "                    if title!=''and body!='':\n",
    "                        self._articles.append((title, body))\n",
    "                    i=i+1\n",
    "            \n",
    "            \n",
    "    def valid_input(self, headline, article_text):\n",
    "        return headline != '' and article_text != ''     \n",
    "        \n",
    "         \n",
    "    def score(self, article):\n",
    "        \"\"\" Assigns each sentence in the document a score\"\"\"\n",
    "\n",
    "        headline = article[0]\n",
    "        sentences = self.split_into_sentences(article[1])\n",
    "        #frequency_scores = self.frequency_scores(article[1])\n",
    "        #let's try a simple query\n",
    "        for i, s in enumerate(sentences):\n",
    "            score=self.sentence_similarity(s, 'What was the original source of COVID-19')\n",
    "            if score is not None:\n",
    "                self._scores[s] = score\n",
    "              \n",
    "    def generate_summaries(self):\n",
    "        \n",
    "        self.dict_ = {'title':[],'body_text': []}\n",
    "\n",
    "        \"\"\" If article is shorter than the desired summary, just return the original articles.\"\"\"\n",
    "       \n",
    "        for article in self._articles:\n",
    "            total_num_sentences = 0\n",
    "            total_num_sentences += len(self.split_into_sentences(article[1]))\n",
    "        \n",
    "            self._scores = Counter()\n",
    "            self.score(article)\n",
    "            highest_scoring = self._scores.most_common(SUMMARY_LENGTH)\n",
    "            print(\"## Headlines: \")\n",
    "            print(\"- \" + article[0])\n",
    "            print('*body**')\n",
    "        # Appends highest scoring \"representative\" sentences, returns as a single summary paragraph.\n",
    "            summr=' '.join([sent[0] for sent in highest_scoring])\n",
    "            print(summr)\n",
    "            self.dict_['body_text'].append(summr)\n",
    "            self.dict_['title'].append(article[0])\n",
    "        self.papers = pd.DataFrame(self.dict_, columns=['title','body_text'])\n",
    "        return self.papers\n",
    "       \n",
    "\n",
    "    def remove_smart_quotes(self, text):\n",
    "       \n",
    "        \"\"\" Only concerned about smart double quotes right now. \"\"\"\n",
    "        #text=re.sub(\"([\\(\\[].*?[\\)\\]][\\(\\[].*?[\\)\\]]+)\", ' ', text)\n",
    "        text=re.sub(\"[\\(\\[].*?[\\)\\]]\", '', text)\n",
    "        \n",
    "        try:\n",
    "            url = re.search(r'((https*:\\/*)([^\\/\\s]+))(.[^\\s]+)', text)\n",
    "            repl_url = url.group(3)\n",
    "            text = re.sub(r'((https*:\\/*)([^\\/\\s]+))(.[^\\s]+)',repl_url, text)\n",
    "        except:\n",
    "            pass #there might be emails with no url in them\n",
    "        #text=re.sub(\"[\\[]()*?[\\]]\", \"\", text)#remove in-text citation\n",
    "        \n",
    "        text=re.sub(r\"[<>()(,)|&©ø\\[\\]\\'\\\";?~*!]\", ' ', text) #remove <>()|&©ø\"',;?~*!\n",
    "        text=re.sub(\"(\\\\t)\", ' ', text) #remove escape charecters\n",
    "        text=re.sub(\"(\\\\r)\", ' ', text) \n",
    "        text=re.sub(\"(\\\\n)\", ' ', text)\n",
    "        text= re.sub(\"(\\s+)\",' ',text) #remove multiple spaces\n",
    "        text=re.sub(\"doi.org doi:\", ' ',text)\n",
    "        text=re.sub(\"doi\", ' ',text)\n",
    "        \n",
    "        return text\n",
    "\n",
    "\n",
    "    def split_into_sentences(self, text):\n",
    "        tok = nltk.data.load('tokenizers/punkt/english.pickle')\n",
    "        sentences = tok.tokenize(self.remove_smart_quotes(text))\n",
    "        sentences = [sent.replace('\\n', '') for sent in sentences if len(sent) > 10]       \n",
    "        for sent in sentences:\n",
    "            if 'doi' in sent:\n",
    "                sent.replace(sent, '')\n",
    "\n",
    "        return sentences\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "## Headlines: \n",
      "- title\n",
      "*body**\n",
      "\n",
      "## Headlines: \n",
      "- The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3\n",
      "*body**\n",
      "In addition these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. Therefore it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. However replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. As near wt level of 353 replication was observed when only one PK was present all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. It is unclear why some viral isolates 483 contain two three or four PKs is still unknown but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint is the . bioRxiv preprint The copyright holder for this preprint is the . Significance shown is compared to wt replicon. As has been described with Mengo virus it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system . The evidence 305 for loop-downstream interaction was weaker for PK4. Contigs that matched the FMDV 227 library were assembled 228 author/funder. can occur in the absence of PKs at least one is required for wild-type replication.\n",
      "## Headlines: \n",
      "- Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China\n",
      "*body**\n",
      "Xinyang Jiujiang Nanchang and Yiyang are medium-sized cities neighboring Hubei. However this is far from adequate for Wuhan given the severity of the epidemic. The epidemiological parameters may have changed after January 23 2020 when the lockdown measure was implemented. Note that the cities in Hubei are not presented here because of their significantly higher vulnerability as compared to others. The ten most vulnerable non-Hubei cities are also listed in the figure. This model was then used to estimate IDi the number of infected individuals in each city. Our analysis revealed that several medium-sized cities particularly those neighboring Hubei are highly vulnerable due to the lack of healthcare resources. In particular Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources. In addition to Wuhan other cities in Hubei are in need of urgent external help as well with respect to health resources as multiple localized outbreaks are taking place. Recent studies indicated that the confirmed case data for Hubei Province in particular Wuhan was subject to the under-reporting bias while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . For the identified cities with high vulnerability it is important to increase the reserve of healthcare resources or have contingent plans. So far there have been over 11 000 non-local healthcare professionals mobilized by the public health authorities from other parts of China fighting against the epidemic in Wuhan.\n",
      "## Headlines: \n",
      "- Real-time, MinION-based, amplicon sequencing for lineage typing of infectious bronchitis virus from upper respiratory samples\n",
      "*body**\n",
      "One area that is problematic for a test such as AmpSeq is to detect all the isolates that may be present. Thus it is another tool for the detection of IBV especially in cases when multiple isolates may be present or when genotyping is especially important. 7 IB is often complicated by secondary bacterial and viral infections . These samples were previously tested for IBV and GA08 serotype of IBV. Each of these read sub-clusters potentially represents a different isolate and was further used for interpretation. Based on the two conventional RT-PCRs and the GA08 RT-qPCR it was determined that only samples #8 and #9 contained a single IBV isolate while the other 7 samples were positive for 2-4 types of IBV . Samples with pathogenic IBV strains were tested with the pan-IBV RT-qPCR and all five samples were IBV positive . bioRxiv preprint Five samples from Run 1 were tested with a pan-IBV RT-qPCR assay 4 and four of those were positive for IBV. DE072 was detected in samples #1 and #9 as the lone GIV-1 isolate but was also detected in three samples that also contained GA98 . The copyright holder for this preprint is the author/funder. All 5 samples were positive for IBV MDL_DMV1639 variant of IBV . Sequencing data was further analyzed to determine if multiple IBV genotypes or lineages could be detected.\n",
      "## Headlines: \n",
      "- A Combined Evidence Approach to Prioritize Nipah Virus Inhibitors\n",
      "*body**\n",
      "NVIK is freely accessible at: vinodscaria.rnabiology.org . It is made available under a The copyright holder for this preprint is the . It can be observed that most compounds in NVIK are cyclic in nature. There may be common structural features in these compounds that are responsible for their activity against NiV . Some of these compounds are available in DrugBank and are predicted to have anti-nipah properties. and the clusters are also reported in CIRCOS. For example GS-441524 is a near neighbour of GS-441524 . Despite being a lethal pathogen there is not a single drug under clinical trial against this virus. This illustrates that the NVIK library is chemically diverse. In total there are 34 NVIs similar to 76 FDA approved drugsf. Of these EC50/IC50 values could be obtained for 70 compounds and structures are available for all 98 inhibitors. Currently there is no known treatment for the disease.\n",
      "## Headlines: \n",
      "- Assessing spread risk of Wuhan novel coronavirus within and beyond China, January-April 2020: a travel network-based modelling study\n",
      "*body**\n",
      "Two Baidu datasets were used in this study. It has been determined that the pathogen causing the viral pneumonia among affected individuals is a new coronavirus . is the The copyright holder for this preprint . The main destinations were presented by region in the supplemental materials . According to Wuhan authorities it is . Of the confirmed cases 2788 are severe and 426 people have died. Nevertheless it is important to note that our study has several major limitations. The case data are available from Dr. Shengjie Lai . If Wuhan s lockdown had not have been undertaken our analyses suggest the main destination cities of population outflow since LNY would have been similar to the situation two weeks prior to LNY . For example exit and entry screening may be futher extended to capture travellers with fever who have potentially been exposed to the virus in high-risk Chinese cities where local transmission has been established. Other places with high risks were Beijing Shanghai Guangzhou and other large cities. A total of 164 cases with available data as of January 25th 2020 were included.\n",
      "## Headlines: \n",
      "- TWIRLS, an automated topic-wise inference method based on massive literature, suggests a possible mechanism via ACE2 for the pathological changes in the human host after coronavirus infection\n",
      "*body**\n",
      "It is also involved in the hydrolysis and activation of hemagglutinin proteins which are essential for viral infectivity . .org 24.20025437  : medRxiv preprint CSHGs in these networks were associated with C3 or C10 categories and five genes were associated with both. The relative position of any CSHG to a certain CSSE can be estimated by TWIRLS . The copyright holder for this preprint is . Using TWIRLS we were able to identify both ACE2 and DPP4 genes as CSHGs and both were significantly associated with the C5 category. The dataset used in this pipeline were from PubMed articles. In addition to ACE2 and DPP4 other CSHGs that are significantly associated with the C5 category include FURIN and TMPRSS2. The IFN-γ-inducible protein 10 is mainly expressed in the lung and is a chemoattractant for activated T cells. The coronavirus study-specific entities were then identified in only the clean texts containing CSHGs. Although Ang II was originally described as an effective vasoconstrictor there is growing evidence that it is closely involved in the inflammatory response of the immune system. MISC indicates the label cannot be summarized. Compared to random genes the regulatory connections between CSHGs were significantly enriched .\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "## Headlines: \n",
      "- Title: Viruses are a dominant driver of protein adaptation in mammals\n",
      "*body**\n",
      "First there may still be many undiscovered VIPs. This is most The copyright holder for this preprint is the author/funder. It is also plausible that a substantial proportion of the The copyright holder for this preprint is the author/funder. Nonetheless given that many VIPs were discovered only . Since VIPs are more 23 constrained than non-VIPs and tend to have more non-synonymous deleterious low 24 frequency variants than non-VIPs is the author/funder. The copyright holder for this preprint is the author/funder. CC-BY 4.0 International license is made available under a The copyright holder for this preprint is the author/funder. However it is unknown whether the war against viruses is fought by a 6 professional army of specifically antiviral proteins or whether it is a global war fought 7 by a broad range of VIPs. In all the 20 GO categories pN/pS is lower 3 in VIPs than in non-VIPs and the difference is significant for 17 of these categories 4 . bioRxiv preprint 7 The classic MK test is known to be biased downward by the presence of slightly 1 deleterious non-synonymous variants and this bias is difficult to eliminate fully even by 2 excluding low frequency variants . Within the past few years there has 20 been no sign that the pace of discovery of new VIPs is slowing down . The stronger purifying selection acting on VIPs 1 is apparent within most functions.\n",
      "## Headlines: \n",
      "- The impact of regular school closure on seasonal influenza epidemics: a data-driven spatial transmission model for Belgium\n",
      "*body**\n",
      "is the author/funder. This was also found in previous empirical studies . Travel is known to be responsible for the spatial dissemination of influenza . The second is the mobility of individuals. The resulting agespecific reductions for mobility are defined in Table 1 . Mixing is patch-dependent and determined by the local demography. Though the simulated dynamics is spatially explicit calibration was performed on Brussels district only i.e. The copyright holder for this preprint is the author/funder. The reduction is expected to be heterogeneous across mixing groups because of compensatory behaviors . In all these cases the start of the epidemic is the only aspect that is being altered whereas the school calendar remains fixed. This may be due to the model neglecting the mobility coupling between these regions and the neighboring countries which is considerably large in some districts . This is generally not reported by influenza surveillance systems as data are collected on a weekly basis.\n",
      "## Headlines: \n",
      "- Carbon Nanocarriers Deliver siRNA to Intact Plant Cells for Efficient Gene\n",
      "*body**\n",
      "RNAse A was purchased from Takara Bio. Significance is measured with a two-tailed unpaired t-test. We have shown that 100 nM siRNA on SWNTs is an optimal dose to be used in mGFP5 silencing studies . N = 10 technical replicates were imaged. Next 50 μL of 0.05 mg/mL NeutrAvidin was added followed by 50 μL of 1.0 mg/L RNA-SWNT which non-specifically adsorbs to NeutrAvidin. The copyright holder for this preprint is the author/funder. bioRxiv preprint The copyright holder for this preprint is the author/funder. UltraPure DNase/RNase-free distilled water from Invitrogen was used for qPCR and EMD Millipore Milli-Q water was used for all other experiments. An annealing temperature of 60˚C and 40 cycles were used for qPCR. 10 μM 5 labelled Cy3-RNA was added to an equal mass of SWNTs. For each sample N = 3 replicates are 3 channels on a microfluidic slide that were prepared independently. The a-antisense siRNA strand was utilized in this assay.\n",
      "## Headlines: \n",
      "- Protective Population Behavior Change in Outbreaks of Emerging Infectious Disease 1 2\n",
      "*body**\n",
      "It is made available under a The copyright holder for this preprint is the . While the isolation of 46 infected individuals is beneficial and should be encouraged it challenges data analysts because 47 it is time-varying and reflects dynamic and often unpredictable human behavior. 256 Furthermore the quality of these estimates will be impacted by the metric chosen as seen by 257 the notable but distinct trends detected in DSOH and MRR. We identified little difference in the mean change of 196 the MRR for the Ebola outbreak depending on predictor While our findings demonstrate large and statistically significant differences in MRR it is 231 notable that the calculated rates of change in the MRRs are within a factor of ten to seven of each other 233 with the mean change being the lowest in the EVD outbreak in Liberia and the highest 234 in the MERS outbreak in South Korea. Similarly while large disparities in DSOH are obvious these data highlight that all societies quickly adapt to outbreaks of emerging infections. The three MERS outbreaks did not differ 175 significantly from one another and had limited precision . The mean change 169 of the MRR in the 170 Ebola epidemic in Lofa 171 County Liberia was 172 significantly higher than the mean change of the MRR for the overall Liberian epidemic and the 173 . 13 While infection prevention 63 often includes vaccination progress to develop effective vaccines for emerging infections is 64 slow and not necessarily more effective than isolation of infected individuals. estimation is complicated however by efforts to isolate infected individuals in hospitals or 45 other settings to decrease contact with the susceptible population. We have shown that public health practices for isolating infected individuals from the 249 susceptible population vary significantly by pathogen and location but can in some cases be 250 predicted by the timing and serial interval of the epidemic. This study detected variation in 251 DSOH and MRR based on epidemic location and outbreak type indicating that it may be 252 possible to estimate a general range of the rate of change in these variables over time. 17 68 69 The focus of this paper is the identification of key similarities and differences in the behavioral 70 response to outbreaks of three emerging zoonotic infections.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>body_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>title</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "      <td>In addition these differences were more pronou...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "      <td>Xinyang Jiujiang Nanchang and Yiyang are mediu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing f...</td>\n",
       "      <td>One area that is problematic for a test such a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A Combined Evidence Approach to Prioritize Nip...</td>\n",
       "      <td>NVIK is freely accessible at: vinodscaria.rnab...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Assessing spread risk of Wuhan novel coronavir...</td>\n",
       "      <td>Two Baidu datasets were used in this study. It...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TWIRLS, an automated topic-wise inference meth...</td>\n",
       "      <td>It is also involved in the hydrolysis and acti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Title: Viruses are a dominant driver of protei...</td>\n",
       "      <td>First there may still be many undiscovered VIP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>The impact of regular school closure on season...</td>\n",
       "      <td>is the author/funder. This was also found in p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Carbon Nanocarriers Deliver siRNA to Intact Pl...</td>\n",
       "      <td>RNAse A was purchased from Takara Bio. Signifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Protective Population Behavior Change in Outbr...</td>\n",
       "      <td>It is made available under a The copyright hol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                title  \\\n",
       "0                                               title   \n",
       "1   The RNA pseudoknots in foot-and-mouth disease ...   \n",
       "2   Healthcare-resource-adjusted vulnerabilities t...   \n",
       "3   Real-time, MinION-based, amplicon sequencing f...   \n",
       "4   A Combined Evidence Approach to Prioritize Nip...   \n",
       "5   Assessing spread risk of Wuhan novel coronavir...   \n",
       "6   TWIRLS, an automated topic-wise inference meth...   \n",
       "7   Title: Viruses are a dominant driver of protei...   \n",
       "8   The impact of regular school closure on season...   \n",
       "9   Carbon Nanocarriers Deliver siRNA to Intact Pl...   \n",
       "10  Protective Population Behavior Change in Outbr...   \n",
       "\n",
       "                                            body_text  \n",
       "0                                                      \n",
       "1   In addition these differences were more pronou...  \n",
       "2   Xinyang Jiujiang Nanchang and Yiyang are mediu...  \n",
       "3   One area that is problematic for a test such a...  \n",
       "4   NVIK is freely accessible at: vinodscaria.rnab...  \n",
       "5   Two Baidu datasets were used in this study. It...  \n",
       "6   It is also involved in the hydrolysis and acti...  \n",
       "7   First there may still be many undiscovered VIP...  \n",
       "8   is the author/funder. This was also found in p...  \n",
       "9   RNAse A was purchased from Takara Bio. Signifi...  \n",
       "10  It is made available under a The copyright hol...  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import csv\n",
    "f= open(\"summr.csv\", encoding=\"utf-8-sig\")\n",
    "reader = csv.reader(f, delimiter=',')\n",
    "magic = Summarizer(reader)\n",
    "magic.generate_summaries()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
